src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs

```diff
@@ -395,7 +451,62 @@ private static void ProcessResult(IAsyncResult asyncResult, AsyncCallback asyncC
             HttpInstrumentationEventSource.Log.FailedProcessResult(ex);
         }
 
-        activity.Stop();
+        activity?.Stop();
+
+        if (HttpClientDuration.Enabled)
+        {
+            double durationMs;
+            if (activity != null)
+            {
+                durationMs = activity.Duration.TotalMilliseconds;
+            }
+            else
+            {
+                var endTimestamp = Stopwatch.GetTimestamp();
+                var durationS = (endTimestamp - startTimestamp) / (double)Stopwatch.Frequency;
+                durationMs = durationS * 1000;
+            }
+
+            TagList tags = default;
+
+            if (metricsEmitOldAttributes)
+            {
+                tags.Add(SemanticConventions.AttributeHttpFlavor, HttpTagHelper.GetFlavorTagValueFromProtocolVersion(request.ProtocolVersion));
+                tags.Add(SemanticConventions.AttributeHttpMethod, request.Method);
+                tags.Add(SemanticConventions.AttributeHttpScheme, request.RequestUri.Scheme);
+                tags.Add(SemanticConventions.AttributeNetPeerName, request.RequestUri.Host);
+                if (!request.RequestUri.IsDefaultPort)
+                {
+                    tags.Add(SemanticConventions.AttributeNetPeerPort, request.RequestUri.Port);
+                }
+            }
+
+            if (metricsEmitNewAttributes)
+            {
+                tags.Add(SemanticConventions.AttributeHttpRequestMethod, request.Method);
+                tags.Add(SemanticConventions.AttributeServerAddress, request.RequestUri.Host);
+                tags.Add(SemanticConventions.AttributeNetworkProtocolVersion, HttpTagHelper.GetFlavorTagValueFromProtocolVersion(request.ProtocolVersion));
+                if (!request.RequestUri.IsDefaultPort)
+                {
+                    tags.Add(SemanticConventions.AttributeServerPort, request.RequestUri.Port);
+                }
+            }
+
+            if (httpStatusCode.HasValue)
+            {
+                if (metricsEmitOldAttributes)
+                {
+                    tags.Add(SemanticConventions.AttributeHttpStatusCode, (int)httpStatusCode.Value);
+                }
+
+                if (metricsEmitNewAttributes)
+                {
+                    tags.Add(SemanticConventions.AttributeHttpResponseStatusCode, (int)httpStatusCode.Value);
+                }
+            }
+
+            HttpClientDuration.Record(durationMs, tags);
```

Comment:
We need to use different histogram based on the configrued option

metricsEmitOldAttributes - > `http.client.duration`
metricsEmitNewAttributes -> `http.client.request.duration`

Reply:
I think this PR is focused on adding metrics to NetFx, where my PR was adding the NEW metric to the library. I don't think we should try to accomplish both goals in a single PR. :)

Reply:
@TimothyMothra I think it is ok to do this here as your PR only updates `HttpHandlerMetricsDiagnosticListener`. So it is ok to do this here. But we need two different Histograms for this to be correct.

Reply:
If the style is "dupe" what happens? You get both old + new histograms each with the full set of tags? Or you get old histogram with the old tags and new histogram with the new tags?

Reply:
>  you get old histogram with the old tags and new histogram with the new tags

This^

Reply:
These HttpClient tests currently are exercising both the .NET Framework & .NET/Core style instrumentations. If we make this change here the tests will start failing for the .NET/Core style. I'm thinking we should tackle both on @TimothyMothra's PR?

Reply:
Could we remove the new conventions from here then?

Reply:
> Could we remove the new conventions from here then?

I just pulled @TimothyMothra's work from #4891 into this PR. Seemed like more work to remove it all and then put it back in.

Reply:
> I just pulled @TimothyMothra's work from #4891 into this PR. 

😲